pandas/core/arrays/categorical.py

```diff
@@ -1882,17 +1882,11 @@ def __getitem__(self, key):
         """
         Return an item.
         """
-        if isinstance(key, (int, np.integer)):
-            i = self._codes[key]
-            return self._box_func(i)
-
-        key = check_array_indexer(self, key)
-
-        result = self._codes[key]
-        if result.ndim > 1:
+        result = super().__getitem__(key)
+        if np.ndim(result) > 1:
```

Comment:
Is `np.ndim(result) > 1` generally slower than something like `not lib.is_scalar(result) and result.ndim > 1`?

result = 5
%timeit np.ndim(result) > 1
2.77 µs ± 48.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

%timeit not lib.is_scalar(result) and result.ndim > 1
154 ns ± 1.2 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)


or is this not important?

Reply:
wow that's neat

Reply:
> not lib.is_scalar(result) and result.ndim > 1?

The trouble here is if we have a tuple result.  I'll take a look at finding a more performant solution.

Reply:
I did wonder about that. but pandas\tests\arrays\categorical and pandas\tests\extension\test_categorical.py didn't fail with this change. Do we have tests for this.

also is it possible to get a nested tuple result? np.ndim would give 2 in this case?

Reply:
Looks like `getattr(result, "ndim", 1) > 1` outperforms both np.ndim and lib.is_scalar checks, and dont need to worry about tuples.  Will update.

> also is it possible to get a nested tuple result? np.ndim would give 2 in this case?

Yes.  Supporting list-like object-dtype is a massive PITA.